src/models.ts

```diff
@@ -297,3 +304,33 @@ export class AdvancedFilter extends Filter {
     return filter;
   }
 }
+
+export interface IDataReference {
+  target: IFilterTarget;
+}
+
+export interface IEqualsDataReference extends IDataReference {
+  equals: string | boolean | number;
+}
+
+export interface IBetweenDataReference extends IDataReference {
+  between: (string | boolean | number)[];
+}
+
+export interface IValueDataReference extends IDataReference {
+  value: string | boolean | number;
+}
+
+export interface IIdentityValue<T> {
```

Comment:
Also, I wasn't quite sure what to name this interface.  It's just a container for an Identity and a Value so I called it `IdentityValue` but looking for better suggestions if there a more semantic name.